lib/ecto/adapters/clickhouse/migration.ex

```diff
@@ -108,14 +108,59 @@ defmodule Ecto.Adapters.ClickHouse.Migration do
     [[drop | @conn.quote_table(index.prefix, index.name)]]
   end
 
-  def execute_ddl({command, %Constraint{} = _constraint})
+  def execute_ddl({command, %Constraint{} = constraint})
       when command in [:create, :create_if_not_exists] do
-    raise "TODO"
+    if constraint.comment do
+      raise "Clickhouse adapter does not support comments on check constraints"
+    end
+
+    unless constraint.validate do
+      raise "Clickhouse adapter does not support check constraints without validation on creation"
```

Comment:
@senconscious 👋

I've been refactoring migrations and noticed 

> Constraint check will not be executed on existing data if it was added.

on https://clickhouse.com/docs/en/sql-reference/statements/alter/constraint

Reply:
Doesn't it mean that we should have the opposite check?

```elixir
if constraint.validate do
  raise "ClickHouse does not support check constraints with validation on creation"
end
```